static/app/components/core/button/index.tsx

```diff
@@ -110,6 +112,165 @@ interface ButtonPropsWithAriaLabel extends BaseButtonProps {
+            <Icon size={size} hasChildren={hasChildren}>
+              <IconDefaultsProvider size={ICON_SIZES[size]}>
+                {props.icon}
+              </IconDefaultsProvider>
+            </Icon>
+          )}
+          {props.children}
+        </ButtonLabel>
+      </StyledButton>
+    </Tooltip>
+  );
+}
+
+export const StyledButton = styled('button')<ButtonProps>`
+  ${p => (p.theme.isChonk ? getChonkButtonStyles(p as any) : getButtonStyles(p as any))}
+`;
+
+export interface LinkButtonProps
+  extends CommonButtonProps,
+    Omit<LinkElementProps, 'role'> {
+  /**
+   * Determines if the link is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Determines if the link is external and should open in a new tab.
+   */
+  external?: boolean;
+  href?: string;
+  /**
+   * If true, the link will not reset the scroll position of the page when clicked.
+   */
+  preventScrollReset?: boolean;
+  /**
+   * Determines if the link should replace the current history entry.
+   */
+  replace?: boolean;
+  to?: string | LocationDescriptor;
+}
+
+export function LinkButton({
+  size = 'md',
+  to,
+  href,
+  disabled,
+  tooltipProps,
+  ...props
+}: LinkButtonProps) {
+  const {handleClick, hasChildren, accessibleLabel} = useButtonFunctionality({
+    ...props,
+    to,
+    href,
+    disabled,
+  });
+
+  return (
+    <Tooltip skipWrapper {...tooltipProps} title={props.title} disabled={!props.title}>
+      <StyledLinkButton
+        aria-label={accessibleLabel}
+        aria-disabled={disabled}
+        size={size}
+        href={disabled ? undefined : href}
+        to={disabled ? undefined : to}
+        disabled={disabled}
+        {...props}
+        onClick={handleClick}
+      >
+        {props.priority !== 'link' && (
+          <InteractionStateLayer
+            higherOpacity={
+              props.priority && ['primary', 'danger'].includes(props.priority)
+            }
+          />
+        )}
+        <ButtonLabel size={size} borderless={props.borderless}>
+          {props.icon && (
+            <Icon size={size} hasChildren={hasChildren}>
+              <IconDefaultsProvider size={ICON_SIZES[size]}>
+                {props.icon}
+              </IconDefaultsProvider>
+            </Icon>
+          )}
+          {props.children}
+        </ButtonLabel>
+      </StyledLinkButton>
+    </Tooltip>
+  );
+}
+
+const StyledLinkButton = styled(
+  ({size: _size, title: _title, external, ...props}: LinkButtonProps) => {
+    if (props.to) {
+      return <Link {...props} to={props.to} role="button" />;
+    }
+
+    if (props.href) {
+      return (
+        <a
+          {...props}
+          {...(external ? {target: '_blank', rel: 'noreferrer noopener'} : {})}
+          role="button"
+        />
+      );
+    }
+
+    const {replace: _replace, preventScrollReset: _preventScrollReset, ...rest} = props;
+    // @ts-expect-error we are spreading anchor link props on a button
+    return <button {...rest} role="button" />;
```

Comment:
this shouldn’t be necessary, as we should require either `to` or `href` being passed as props.

Reply:
Yes, but there are a ton of tests that fail if I remove it because the old button implementation was allowing it. I plan to address it separately.